packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -2771,12 +2771,12 @@ class DiagnosticsProperty<T> extends DiagnosticsNode {
     _valueComputed = true;
     assert(_computeValue != null);
     try {
-      _value = _computeValue!();
+      _value = WeakValue<T>(_computeValue!(), keepAlive: true);
```

Comment:
making this value be kept alive could interfere with the invariant that DiagnosticsNode objects do not keep alive the value they were computed for. Would it be reasonable to instead recompute the value if the previous value has been GCed? Could be worth auditing the Flutter code to see where this is used to evaluate whether that is a concern. On the other hand the _computeValue closures may be themselves keeping objects alive causing the leaks you are looking to avoid so that part would need to be audited as well.